### PR TITLE
Refactor searchlogs output and strip out 0x from addresses

### PR DESCRIPTION
**Pull request by dwalintukan**

### CODE REVIEW

File: src/formatter.js [hunk 1]
```diff
@@ -1,64 +1,87 @@
...
-        const rawlogs = _.map(resultEntry.log, (logEntry) => ({
-          address: Utils.formatHexStr(logEntry.address),
-          data: Utils.formatHexStr(logEntry.data),
-          topics: _.map(logEntry.topics, Utils.formatHexStr)
-        }));
-
-        formatted.log = logDecoder(rawlogs);
+        _.each(resultEntry.log, (item, index) => {
+          const topicHash = item.topics[0];
+          const eventTopicObj = _.find(EventMetadata, { eventHash: topicHash });
```
Comment (frankobe) [inline src/formatter.js]: this will mk qweb.js bodhi specific rather than generic

File: src/event_metadata.js [hunk 2]
```diff
@@ -0,0 +1,47 @@
+module.exports = {
```
Comment (frankobe) [inline src/event_metadata.js]: this is bodhi specific abi

Comment (dwalintukan) [reply src/event_metadata.js]: removed

File: src/formatter.js (hunk 1 above)
Comment (dwalintukan) [reply src/formatter.js]: fixed, remove all bodhi specific logic

File: src/formatter.js [hunk 3]
```diff
@@ -1,64 +1,98 @@
...
-
-        formatted.log = logDecoder(rawlogs);
+        _.each(resultEntry.log, (item, index) => {
+          const eventHash = item.topics[0];
+
+          let eventName;
+          let metadataObj;
+          _.each(contractMetadata, (contractItem, index) => {
+            eventName = (_.invert(contractItem))[eventHash]; 
+
```
Comment (chrisli30) [inline src/formatter.js]: Add comments to this logic

File: src/decoder.js [hunk 4]
```diff
@@ -0,0 +1,20 @@
+const Web3Utils = require('web3-utils');
+const bs58 = require('bs58');
+
+class Decoder {
+
+  static toQtumAddress(hexAddress) {
+    if (!hexAddress) {
+      throw new Error(`hexAddress should not be undefined.`);
+    }
```
Comment (chrisli30) [inline src/decoder.js]: a small thing, but empty string will also go in to this case !hexAddress. Update to typeof hexAddress === undefined or error message 'not be undefined or empty'

File: src/formatter.js (hunk 3 above)
Comment (dwalintukan) [reply src/formatter.js]: old logic, removed

File: src/decoder.js (hunk 4 above)
Comment (dwalintukan) [reply src/decoder.js]: fixed, good call 👍

**Pull request merged**